<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>780</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="RVWidget" name="widgetRV">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>300</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="animationButton">
          <property name="text">
           <string>Animation</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDial" name="fov">
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>120</number>
          </property>
          <property name="singleStep">
           <number>5</number>
          </property>
          <property name="value">
           <number>45</number>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_skybox">
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:10pt; font-style:italic;&quot;&gt;Contexte&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="skybox_Box">
          <item>
           <property name="text">
            <string>Fôret hiver</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Parc</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Château</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Rivière</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Fôret </string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="OGLSettings">
          <property name="minimumSize">
           <size>
            <width>140</width>
            <height>0</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <italic>true</italic>
           </font>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="title">
           <string>Réglages de rendu</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label_selection">
             <property name="text">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:8pt; font-weight:700;&quot;&gt;Sélection objet&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="Object_selected_Box">
             <item>
              <property name="text">
               <string>Dé</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Plane</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Terre</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Torus</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="wireframeCheck">
             <property name="text">
              <string>Fil de fer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cullingCheck">
             <property name="text">
              <string>Culling</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_size">
             <property name="text">
              <string>Taille</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>50</number>
             </property>
             <property name="pageStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickInterval">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="remove_Button">
             <property name="text">
              <string>Supprimer Objet</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="minimumSize">
           <size>
            <width>140</width>
            <height>280</height>
           </size>
          </property>
          <property name="font">
           <font>
            <italic>true</italic>
           </font>
          </property>
          <property name="title">
           <string>Nouveau objet</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <widget class="QPushButton" name="boutonCree">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>250</y>
             <width>80</width>
             <height>24</height>
            </rect>
           </property>
           <property name="text">
            <string>Créer l'objet</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_x">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>60</y>
             <width>111</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;x&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_y">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>90</y>
             <width>111</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;y&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_z">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>120</y>
             <width>111</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;z&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="valeur_x">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>60</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="valeur_y">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>90</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="valeur_z">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>120</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="valeur_roulis">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>150</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="valeur_tangage">
           <property name="geometry">
            <rect>
             <x>61</x>
             <y>180</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="valeur_lacet">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>210</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="label_roulis">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>150</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;roulis&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_tangage">
           <property name="geometry">
            <rect>
             <x>8</x>
             <y>180</y>
             <width>51</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;tangage&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_lacet">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>210</y>
             <width>49</width>
             <height>16</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;lacet&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
          <widget class="QComboBox" name="valeur_type">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>30</y>
             <width>71</width>
             <height>24</height>
            </rect>
           </property>
           <item>
            <property name="text">
             <string>Cube</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Dé</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Planet</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Torus</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Plane</string>
            </property>
           </item>
          </widget>
          <widget class="QLabel" name="label_x_2">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>41</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:700;&quot;&gt;type&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RVWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>rvwidget.h</header>
   <slots>
    <slot>startAnimation()</slot>
    <slot>changeFov(int)</slot>
    <slot>changeOpacity(int)</slot>
    <slot>changeWireframe()</slot>
    <slot>changeCulling()</slot>
    <slot>changeSize(int)</slot>
    <slot>changeSaturation(int)</slot>
    <slot>createCube()</slot>
    <slot>changeSkyBox(int)</slot>
    <slot>changeSelectedObject(QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>animationButton</sender>
   <signal>clicked()</signal>
   <receiver>widgetRV</receiver>
   <slot>startAnimation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>852</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>560</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeFov(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>914</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>583</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wireframeCheck</sender>
   <signal>clicked()</signal>
   <receiver>widgetRV</receiver>
   <slot>changeWireframe()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>840</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>573</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cullingCheck</sender>
   <signal>clicked()</signal>
   <receiver>widgetRV</receiver>
   <slot>changeCulling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>840</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeSize(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>902</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>boutonCree</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>createObjet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>706</y>
    </hint>
    <hint type="destinationlabel">
     <x>768</x>
     <y>591</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>skybox_Box</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeSkyBox(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>914</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>648</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Object_selected_Box</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeSelectedObject(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>276</y>
    </hint>
    <hint type="destinationlabel">
     <x>920</x>
     <y>514</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>remove_Button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>898</x>
     <y>747</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>createObjet()</slot>
  <slot>changeSelectedObject(int)</slot>
  <slot>removeObject()</slot>
 </slots>
</ui>
